cub/cub/device/dispatch/dispatch_segmented_sort.cuh

```diff
@@ -66,6 +70,49 @@
 
 CUB_NAMESPACE_BEGIN
 
+namespace detail
+{
+
+namespace segmented_sort
+{
+// Type used to index within segments within a single invocation
+using per_invocation_segment_offset_t = unsigned int;
```

Comment:
Question: Should this rather be a sized type like `cuda::std::uint32_t`

Reply:
Yes, I replaced all places that are using segment index type to now use this type alias and made that type alias `cuda::std::uint32_t`.